### PR TITLE
e2e: eg controlplane metrics

### PR DESCRIPTION
**Pull request by zirain**
xhref: https://github.com/envoyproxy/gateway/issues/2088

### CODE REVIEW

File: test/e2e/tests/controlplane.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, ControlPlaneTest)
```
Comment (arkodg) [inline test/e2e/tests/controlplane.go:23]:
```suggestion
	ConformanceTests = append(ConformanceTests, ControlPlaneMetrics)
```

File: test/e2e/testdata/prometheus.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+apiVersion: v1
```
Comment (arkodg) [inline test/e2e/testdata/prometheus.yaml:1]: also being created in https://github.com/envoyproxy/gateway/pull/2108

Comment (zirain) [reply test/e2e/testdata/prometheus.yaml:1]: this one `LoadBalancer` type

**Review by arkodg — APPROVED** [review]

**Review by Xunzhuo — APPROVED** [review]

**Comment by Xunzhuo** [comment]
Thanks.

**Pull request merged**